server/src/main/java/io/spine/server/entity/Transaction.java

```diff
@@ -363,16 +366,28 @@ public final void commit() throws InvalidEntityStateException, IllegalStateExcep
                      ? entity.state()
                      : builder.buildPartial();
         if (initialState.equals(newState)) {
-            commitUnchangedState();
+            updateColumnsAndCommit();
         } else {
-            commitChangedState(newState);
+            commitChangedState(newState, true);
         }
     }
 
     private boolean withPhases() {
         return !phases.isEmpty();
     }
 
+    /**
+     * Propagates entity column values to the entity state and commits a transaction.
+     */
+    private void updateColumnsAndCommit() {
+        S stateWithColumns = stateWithColumns();
+        if (initialState.equals(stateWithColumns)) {
+            commitUnchangedState();
+        } else {
+            commitChangedState(stateWithColumns, false);
```

Comment:
This is not obvious. Why do we not `updateColumns` when the columns have been changed?